terraform/main.tf

```diff
@@ -11,30 +11,16 @@ resource "juju_application" "temporal_worker_k8s" {
   constraints = var.constraints
   config      = var.config
 
-  units = var.units
-}
-
-
-resource "null_resource" "attach_image" {
-  provisioner "local-exec" {
-    # Needed since juju_application resource does not support resource map to specify registry creds
-    # Refactor once https://github.com/juju/terraform-provider-juju/issues/620 resolved
-    command = <<EOT
-      echo "${yamlencode({
-    "registrypath" = var.image.image,
-    "username"     = var.image.registry_username,
-    "password"     = var.image.registry_password,
-})}" > temporal_worker_image.yaml
-
-      juju attach-resource ${var.app_name} temporal-worker-image=temporal_worker_image.yaml
+  resources = var.image.image != "" ? {
+    "temporal-worker-image" = var.image.image
+  } : {}
 
-      rm temporal_worker_image.yaml
-    EOT
-}
-
-depends_on = [resource.juju_application.temporal_worker_k8s]
+  registry_credentials = {
```

Comment:
would we need a condition here as well? if the `image_repository` is empty, do we want to include `registry_credentials`?